File: frontend/data/faculties.ts
const FACULTIES: Record<string, { displayName: string; bgClass: string; description: string; symbol?: string }> = {
  gryffindor: {
    displayName: "Gryffindor",
    bgClass: "bg-red-900 text-white",
    description: "Gryffindor values bravery, daring, nerve, and chivalry.",
    symbol: "🦁",
  },
  slytherin: {
    displayName: "Slytherin",
    bgClass: "bg-green-900 text-white",
    description: "Slytherin values ambition, cunning, leadership, and resourcefulness.",
    symbol: "🐍",
  },
  ravenclaw: {
    displayName: "Ravenclaw",
    bgClass: "bg-blue-900 text-white",
    description: "Ravenclaw values wisdom, intellect, and creativity.",
    symbol: "🦅",
  },
  hufflepuff: {
    displayName: "Hufflepuff",
    bgClass: "bg-amber-400 text-black",
    description: "Hufflepuff values loyalty, patience, and fairness.",
    symbol: "🦡",
  },
};

export default FACULTIES;

File: backend/app/api/faculties.py
from fastapi import APIRouter

router = APIRouter()

FACULTIES = [
    {
        "name": "Gryffindor",
        "traits": ["Bravery", "Courage", "Chivalry"],
        "colors": ["Red", "Gold"]
    },
    {
        "name": "Slytherin",
        "traits": ["Ambition", "Cunning", "Leadership"],
        "colors": ["Green", "Silver"]
    },
    {
        "name": "Ravenclaw",
        "traits": ["Wisdom", "Intellect", "Creativity"],
        "colors": ["Blue", "Bronze"]
    },
    {
        "name": "Hufflepuff",
        "traits": ["Loyalty", "Patience", "Fairness"],
        "colors": ["Yellow", "Black"]
    }
]

@router.get("/")
def get_faculties():
    return FACULTIES
